src/OnlineRetailPortal.MongoDBStore/MongoProductStore.cs

```diff
@@ -111,5 +112,98 @@ public async Task<ProductEntity> UpdateProductAsync(ProductEntity productEntity)
             }
             return mongoEntity.ToModel();
         }
+
+        public async Task<ProductEntity> UpdateProductAsync(UpdateProductEntity updateProductEntity)
+        {
+            MongoEntity getResult;
+            var hello = updateProductEntity;
+            var productStoreCollection = _db.GetCollection<MongoEntity>(_collection);
+            getResult = await productStoreCollection.Find(x => x.Id == updateProductEntity.Id).FirstOrDefaultAsync();
+
+            getResult = UpdateProductEntity(await productStoreCollection.Find(x => x.Id == updateProductEntity.Id).FirstOrDefaultAsync(),updateProductEntity);
+            var filter = Builders<MongoEntity>.Filter.Eq("Id", updateProductEntity.Id);
+            try
+            {
+                var result = await productStoreCollection.ReplaceOneAsync(
+                            filter,
+                            getResult,
+                            new UpdateOptions { IsUpsert = false });
+            }
+            catch
+            {
+                throw new BaseException(int.Parse(ErrorCode.DataBaseDown()), Error.DataBaseDown(), null, HttpStatusCode.GatewayTimeout);
+            }
+            return getResult.ToModel();
+        }
+
+        private MongoEntity UpdateProductEntity(MongoEntity getResult, UpdateProductEntity updateProductEntity)
+        {
+            if (getResult != null)
```

Comment:
find better way if possible instead of if

Reply:
done